python/ray/dashboard/client/src/pages/dashboard/tune/TuneErrors.tsx

```diff
@@ -0,0 +1,131 @@
+import { Theme } from "@material-ui/core/styles/createMuiTheme";
+import createStyles from "@material-ui/core/styles/createStyles";
+import withStyles, { WithStyles } from "@material-ui/core/styles/withStyles";
+import React from "react";
+import { connect } from "react-redux";
+import { StoreState } from "../../../store";
+import { dashboardActions } from "../state";
+import Table from "@material-ui/core/Table";
+import TableBody from "@material-ui/core/TableBody";
+import TableCell from "@material-ui/core/TableCell";
+import TableHead from "@material-ui/core/TableHead";
+import TableRow from "@material-ui/core/TableRow";
+import NumberedLines from "../../../common/NumberedLines";
+import Link from "@material-ui/core/Link";
+import Dialog from "@material-ui/core/Dialog";
+import DialogContent from "@material-ui/core/DialogContent";
+
+const styles = (theme: Theme) =>
+  createStyles({
+    table: {
+      marginTop: theme.spacing(1)
+    },
+    cell: {
+      padding: theme.spacing(1),
+      textAlign: "center",
+      "&:last-child": {
+        paddingRight: theme.spacing(1)
+      }
+    }
+  });
+
+const mapStateToProps = (state: StoreState) => ({
+  tuneInfo: state.dashboard.tuneInfo
+});
+
+const mapDispatchToProps = dashboardActions;
+
+interface State {
+  currentError: string;
+  open: boolean;
+}
+
+class TuneErrors extends React.Component<
+  WithStyles<typeof styles> &
+    ReturnType<typeof mapStateToProps> &
+    typeof mapDispatchToProps,
+  State
+> {
+  state: State = {
+    currentError: "",
+    open: false
+  };
+
+  handleOpen = (key: string) => {
+    this.setState({
+      open: true,
+      currentError: key
+    });
+  };
+
+  handleClose = () => {
+    this.setState({
+      open: false
+    });
+  };
+
+  render() {
+    const { classes, tuneInfo } = this.props;
+    const { currentError, open } = this.state;
+
+    if (tuneInfo === null || Object.keys(tuneInfo["errors"]).length === 0) {
+      return null;
+    }
+
+    return (
+      <div>
```

Comment:
We can use a `<React.Fragment>` here to avoid the redundant `<div>`.

Reply:
Changed